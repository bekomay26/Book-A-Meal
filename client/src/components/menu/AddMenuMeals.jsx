import React from 'react';
import PropTypes from 'prop-types';


const AddMenuMeals = ({ onSave, onSelect, meals }) => (
  <div>
    <form>
      <div className="row">
        <div className="col-12 subm">
          <input type="submit" value="Save Menu" onClick={onSave} />
        </div>
      </div>
      <div className="menu-items">
        <ul id="todayslist">
          {
            meals.map(meal => (
              <li
                key={meal.id}
                onKeyPress={event => onSelect(event, meal)}
                onClick={event => onSelect(event, meal)}
                tabIndex="0"
                role="button"
              >
                {meal.title}<span className="sel-menu-item-price">&#x20A6; {meal.price}</span>
              </li>))
          }
        </ul>
      </div>
    </form>
  </div>
);

AddMenuMeals.propTypes = {
  onSave: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  meals: PropTypes.array.isRequired,
};

export default AddMenuMeals;
