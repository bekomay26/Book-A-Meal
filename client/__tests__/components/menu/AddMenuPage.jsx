import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddMenuMeals from '../../../src/components/menu/AddMenuMeals';
import { getMealsResponse } from '../../__mocks__/mockMeal';


const mealArray = getMealsResponse.meals;
function setup() {
  const props = {
    onSave: jest.fn(),
    onSelect: jest.fn(),
    meals: mealArray,
  };
  return shallow(<AddMenuMeals {...props} />, reactrouter.get());
}

describe('The Add Menu Meals Component', () => {
  it('should render AddMenuMeals component correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
