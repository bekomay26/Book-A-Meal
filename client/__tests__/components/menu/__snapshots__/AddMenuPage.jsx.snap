// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Add Menu Meals Component should render AddMenuMeals component correctly 1`] = `
<div>
  <form>
    <div
      className="row"
    >
      <div
        className="col-12 subm"
      >
        <input
          onClick={[MockFunction]}
          type="submit"
          value="Save Menu"
        />
      </div>
    </div>
    <div
      className="menu-items"
    >
      <ul
        id="todayslist"
      >
        <li
          key="54"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          gfhgh
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            980
          </span>
        </li>
        <li
          key="53"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Ebaaa
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            550
          </span>
        </li>
        <li
          key="51"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Rice
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            1000
          </span>
        </li>
        <li
          key="49"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Basmatic Rice
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            435
          </span>
        </li>
        <li
          key="48"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Gbegiri ati Ewedu
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            900
          </span>
        </li>
        <li
          key="43"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Dodo
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            340
          </span>
        </li>
        <li
          key="42"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Ogbonnax
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            350
          </span>
        </li>
        <li
          key="41"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Ogbonna
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            350
          </span>
        </li>
        <li
          key="38"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          Asanka
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            509
          </span>
        </li>
        <li
          key="1"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          rice
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            450
          </span>
        </li>
        <li
          key="2"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          abka atam
          <span
            className="sel-menu-item-price"
          >
            ₦ 
            430
          </span>
        </li>
      </ul>
    </div>
  </form>
</div>
`;
